src/lib/hash.ts

```diff
@@ -0,0 +1,28 @@
+import type { Map } from 'maplibre-gl-js';
+
+// Adapted from https://github.com/maplibre/maplibre-gl-js/blob/5d7e6d52000a8569ac2308a9aef14c98933eb0d8/src/ui/hash.ts
+export function getViewportHash(map: Map): string {
+  let center = map.getCenter();
+  let zoom = Math.round(map.getZoom() * 100) / 100;
+  // derived from equation: 512px * 2^z / 360 / 10^d < 0.5px
+  let precision = Math.ceil((zoom * Math.LN2 + Math.log(512 / 360 / 0.5)) / Math.LN10);
+  let m = Math.pow(10, precision);
+  let lat = Math.round(center.lat * m) / m;
+  let lng = Math.round(center.lng * m) / m;
+  let hash = `${zoom}/${lat}/${lng}`;
+
+  let bearing = map.getBearing();
+  let pitch = map.getPitch();
+  if (bearing || pitch) {
+    hash += `/${Math.round(bearing * 10) / 10}`;
+  }
+  if (pitch) {
+    hash += `/${Math.round(pitch)}`;
+  }
+  return `#${hash}`;
```

Comment:
MapLibre has some kind of internal state tracking, https://github.com/maplibre/maplibre-gl-js/blob/5d7e6d52000a8569ac2308a9aef14c98933eb0d8/src/ui/hash.ts#L68. I don't understand what it's used for, and it seems totally unnecessary -- the behavior in this PR seems correct to me.

Reply:
Are you referring to the `hashName` thing? I think that's if you want to store multiple pieces of state in the hash, like the map bounds but also something else.  Basically turning the hash into something like a query string with multiple keys and values.

Seems ok to skip IMO, it can be added later if this turns out to be in high demand